File: SDKs/amazonaws.com/redshift/2012-12-01/go/pkg/models/operations/postdescribeclustertracks.go
package operations




type PostDescribeClusterTracksActionEnum string

const (
    PostDescribeClusterTracksActionEnumDescribeClusterTracks PostDescribeClusterTracksActionEnum = "DescribeClusterTracks"
)



type PostDescribeClusterTracksVersionEnum string

const (
    PostDescribeClusterTracksVersionEnumTwoThousandAndTwelve1201 PostDescribeClusterTracksVersionEnum = "2012-12-01"
)


type PostDescribeClusterTracksQueryParams struct {
    Action PostDescribeClusterTracksActionEnum `queryParam:"style=form,explode=true,name=Action"`
    Marker *string `queryParam:"style=form,explode=true,name=Marker"`
    MaxRecords *string `queryParam:"style=form,explode=true,name=MaxRecords"`
    Version PostDescribeClusterTracksVersionEnum `queryParam:"style=form,explode=true,name=Version"`
    
}

type PostDescribeClusterTracksHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    
}

type PostDescribeClusterTracksRequest struct {
    QueryParams PostDescribeClusterTracksQueryParams 
    Headers PostDescribeClusterTracksHeaders 
    Request []byte `request:"mediaType=text/xml"`
    
}

type PostDescribeClusterTracksResponse struct {
    Body []byte 
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/redshift/2012-12-01/typescript/src/sdk/models/operations/postdescribeclustertracks.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";

export enum PostDescribeClusterTracksActionEnum {
    DescribeClusterTracks = "DescribeClusterTracks"
}

export enum PostDescribeClusterTracksVersionEnum {
    TwoThousandAndTwelve1201 = "2012-12-01"
}


export class PostDescribeClusterTracksQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: PostDescribeClusterTracksActionEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=Marker" })
  marker?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=MaxRecords" })
  maxRecords?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: PostDescribeClusterTracksVersionEnum;
}


export class PostDescribeClusterTracksHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class PostDescribeClusterTracksRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: PostDescribeClusterTracksQueryParams;

  @Metadata()
  headers: PostDescribeClusterTracksHeaders;

  @Metadata({ data: "request, media_type=text/xml" })
  request?: Uint8Array;
}


export class PostDescribeClusterTracksResponse extends SpeakeasyBase {
  @Metadata()
  body?: Uint8Array;

  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;
}
